### PR TITLE
Update event schema to comply with the propposed structure

### PR DESCRIPTION
**Pull request by josejulio**
## Describe your changes

- Updates the event being send to use the proposed format

Note that this does not update the stdout event manager. That one takes the receiving object and prints it directly to stdout. We could reuse some logic to also print the same string if desired.

With this PR, the content in kafka from sending [host.json](https://github.com/project-kessel/inventory-api/blob/main/data/host.json):

```json
{
   "metadata":{
      "id":8,
      "last_reported":"2024-09-16T09:41:00.484463105-06:00",
      "resource_type":"rhel-host",
      "workspace":"workspace1",
      "labels":null
   },
   "reporter_data":{
      "reporter_instance_id":"curl/8.6.0",
      "reporter_type":"OCM",
      "last_reported":"2024-09-16T09:41:00.484829048-06:00",
      "local_resource_id":"1",
      "reporter_version":"0.1",
      "console_href":"www.example.com",
      "api_href":"www.example.com"
   },
   "resource_data":null
}
```

to: 

```json
{
   "specversion":"1.0",
   "id":"bef81b33-7476-11ef-9737-a63ce1be7759",
   "source":"http://localhost:8081",
   "type":"redhat.inventory.resources.rhel-host.created",
   "subject":"/resources/rhel-host/1",
   "datacontenttype":"application/json",
   "time":"2024-09-16T21:58:03.056562942Z",
   "data":{
      "metadata":{
         "id":18,
         "last_reported":"2024-09-16T15:58:03.056562942-06:00",
         "resource_type":"rhel-host",
         "workspace":"workspace1",
         "labels":null
      },
      "reporter_data":{
         "reporter_instance_id":"curl/8.6.0",
         "reporter_type":"OCM",
         "last_reported":"2024-09-16T15:58:03.056852412-06:00",
         "local_resource_id":"1",
         "reporter_version":"0.1",
         "console_href":"www.example.com",
         "api_href":"www.example.com"
      },
      "resource_data":null
   }
}
```

## Ticket reference (if applicable)
Fixes #

## Checklist

* [ ] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [ ] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [ ] Are the agreed upon coding/architectural practices applied?

* [ ] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

File: internal/data/k8spolicies/k8spolicies.go [hunk 1]
```diff
@@ -72,7 +72,10 @@ func (r *k8spoliciesRepo) Update(ctx context.Context, model *biz.K8sPolicy, id s
 
 	if r.Eventer != nil {
 		producer, _ := r.Eventer.Lookup(identity, biz.ResourceType, model.ID)
-		evt := eventingapi.NewUpdateEvent(biz.ResourceType, model.Metadata.UpdatedAt, &eventingapi.EventResource[biz.K8sPolicyDetail]{
+		// Todo: model.Metadata.Reporters[0].LocalResourceId is the resourceId of the resource in the scope of the reporter.
+		// When we have multiple reporters we will need to make sure to publish the correct one in the eventing api.
```
Comment (jmelis) [inline internal/data/k8spolicies/k8spolicies.go:76]: I think this is not an issue.

The entry point of this call is always:
https://github.com/project-kessel/inventory-api/blob/508fc28b4cc30ac7a5602cf2337263864f6bc968/api/kessel/inventory/v1beta1/resources/k8s_policies_service.proto#L22-L27

Which has a single reporter and resource data:
https://github.com/project-kessel/inventory-api/blob/508fc28b4cc30ac7a5602cf2337263864f6bc968/api/kessel/inventory/v1beta1/resources/k8s_policy.proto#L15-L23

In other words, `model.Metadata.Reporters` will always be a slice with just one element.

**Review by jmelis — APPROVED** [review]
LGTM

 I'm in agreement with all the decisions in this PR:

- passing a `public_url` configuration option
- separating the event methods between resources and relationships
- leaving figuring out a stable event id for the future
- while I don't like the use of generics for the events, I think we can merge this for now, and iterate on this for a better organization for this

**Review by akoserwal — APPROVED** [review]

File: internal/data/k8spolicies/k8spolicies.go (hunk 1 above)
Comment (josejulio) [reply internal/data/k8spolicies/k8spolicies.go:76]: Yes, but the idea is that multiple reporters can report the same resource. We will then aggregate in the database/orm and will eventually end with multiple reporters (in the future).
see: https://github.com/project-kessel/inventory-api/tree/main/api/kessel/inventory/v1beta1/documentation#resource-deduplication
